src/Eventso.Subscription.Hosting/PoisonEventRetryingHost.cs

```diff
@@ -0,0 +1,111 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Eventso.Subscription.Configurations;
+using Eventso.Subscription.Kafka;
+using Eventso.Subscription.Kafka.DeadLetter;
+using Eventso.Subscription.Kafka.DeadLetter.Postgres;
+using Eventso.Subscription.Kafka.DeadLetter.Store;
+using Eventso.Subscription.Observing.Batch;
+using Eventso.Subscription.Observing.DeadLetter;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace Eventso.Subscription.Hosting
+{
+    public sealed class PoisonEventRetryingHost : BackgroundService
+    {
+        private const long RetryLockId = 1; 
+        
+        private readonly IConnectionFactory _connectionFactory;
+        private readonly ILogger _logger;
+        private readonly IReadOnlyCollection<TopicRetryingService> _topicRetryingServices;
+
+        public PoisonEventRetryingHost(
+            IEnumerable<ISubscriptionCollection> subscriptions,
+            IMessagePipelineFactory pipelineFactory,
+            IMessageHandlersRegistry handlersRegistry,
+            IPoisonEventStore poisonEventStore,
+            IDeadLetterQueueScopeFactory deadLetterQueueScopeFactory,
+            IConnectionFactory connectionFactory,
+            ILoggerFactory loggerFactory)
+        {
+            _connectionFactory = connectionFactory;
+            _logger = loggerFactory.CreateLogger<SubscriptionHost>();
+
+            _topicRetryingServices = (subscriptions ?? throw new ArgumentNullException(nameof(subscriptions)))
+                .SelectMany(x => x)
+                .Where(x => x.EnableDeadLetterQueue)
+                .Select(x => CreateTopicRetryingService(x, pipelineFactory, handlersRegistry, poisonEventStore, deadLetterQueueScopeFactory, loggerFactory))
+                .ToArray();
+        }
+
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            if (_topicRetryingServices.Count == 0)
+            {
+                await Task.Delay(Timeout.InfiniteTimeSpan, stoppingToken);
+                return;
+            }
+            
+            while (!stoppingToken.IsCancellationRequested)
+            {
+                // TODO hide postgres behind some interface
+                await using var connection = _connectionFactory.ReadWrite();
+                await DistributedMonitor.TryEnter(connection, RetryLockId, stoppingToken);
+
+                try
+                {
+                    foreach (var topicRetryingService in _topicRetryingServices)
+                        await topicRetryingService.Retry(stoppingToken);
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogError(ex, $"Dead letter queue retrying failed.");
```

Comment:
my resharper gets angry when it sees period at the end of a log string

plus, you don't need `$` here and in many other places